src/parallel_simulation/network/network_partition.rs

```diff
@@ -1,17 +1,20 @@
 use crate::io::network::IOLink;
 use crate::parallel_simulation::network::link::{Link, LocalLink, SplitInLink, SplitOutLink};
 use crate::parallel_simulation::network::node::Node;
+use crate::parallel_simulation::network::routing_kit_network::RoutingKitNetwork;
 use std::collections::{HashMap, HashSet};
 
 #[derive(Debug)]
 pub struct NetworkPartition {
+    pub routing_kit_network: RoutingKitNetwork,
```

Comment:
I think we have talked about it already, but I still find it slightly weird, that the `NetworkPartition` has a representation of the local partition of the network, but also a representation of the global `RoutingKitNetwork`. Maybe this can be separated again if the preprocessing for the routing is done outside of `Simulation`?

Reply:
If not, this is also okay, until we come up with a better idea :-)

Reply:
That's no problem. It works even without separating the preprocessing and routing.